src/Collector.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace Humbug;
+
+/**
+ * Class collecting all mutants and their results.
+ *
+ * @category   Humbug
+ * @package    Humbug
+ * @copyright  Copyright (c) 2015 Pádraic Brady (http://blog.astrumfutura.com)
+ * @license    https://github.com/padraic/humbug/blob/master/LICENSE New BSD License
+ * @author     Thibaud Fabre
+ */
+class Collector
+{
+
+    /**
+     * @var int Total mutant count collected
+     */
+    private $totalCount = 0;
+
+    /**
+     * @var int Count of mutants not covered by a test case.
+     */
+    private $shadowCount = 0;
+
+    /**
+     * @var int Count of mutants killed by a test case.
+     */
+    private $killedCount = 0;
+
+    /**
+     * @var Mutant[] Mutants killed by a test case.
+     */
+    private $killed = [];
+
+    /**
+     * @var int Count of mutants that timed out.
+     */
+    private $timeoutCount = 0;
+
+    /**
+     * @var Mutant[] Mutants that resulted in a timeout.
+     */
+    private $timeouts = [];
+
+    /**
+     * @var int Count of mutants that triggered an error.
+     */
+    private $errorCount = 0;
+
+    /**
+     * @var Mutant[] Mutants that triggered an error.
+     */
+    private $errors = [];
+
+    /**
+     * @var int Count of mutants that escaped tests.
+     */
+    private $escapeCount = 0;
+
+    /**
+     * @var Mutant[] Mutants that escaped tests.
+     */
+    private $escaped = [];
+
+    /**
+     * Collects a result associated with a mutant.
+     *
+     * @param Mutant $mutant
+     * @param array $result
+     */
+    public function collect(Mutant $mutant, array $result)
```

Comment:
This $result array is from completely different context which means that it may not have 'timeout', or 'successful', or 'error' key.
Next thing I see is that Collector has to know about its structure.

Would it be better if you introduce: 
* pushTimeoutMutant
* pushErrorMutant
* pushKilledMutant
* pushEscapedMutant

You merged it so quickly. I didn`t even noticed :)